src/pool/pool_jemalloc.c

```diff
@@ -329,6 +345,12 @@ static void *je_realloc(void *pool, void *ptr, size_t size) {
         return NULL;
     }
 
+    if (new_ptr != ptr) {
+        VALGRIND_DO_MEMPOOL_ALLOC(pool, new_ptr, size);
+        VALGRIND_DO_MEMPOOL_FREE(pool, ptr);
```

Comment:
missing utils_annotate_memory_unaccessible(ptr)?

Reply:
No, since we don't mark memory as inaccessible under address sanitizer, we don't need to do anything here for sanitizer. We only rely on valgrind here which already marks memory as inaccessible inside MEMPOOL_FREE.